### ISSUE TITLE
Allow more entrypoints to manipulate directus

### ISSUE THREAD

**Issue by jooola** [directus/api#1623]
I had an idea while thinking how to enhance my development / shipping workflow with directus.

The idea is simply to allow other ways to manipulate the Directus database. Currently the only way to build collections and fields is through the API so mainly through the Vue App.

Some other ways could be using CLI commands (development, maintenance), or using some structured yaml file to describe the collections and fields (deployment, development, backups).

For example a CLI command could load or export a structure definitions in YAML.

Since the project is moving to Laravel, I hoped this could be taken into count, and build some kind of **service layer** before plugin the whole thing into HTTP handlers.

I know Laravel has some really nice features like [Thinker](https://laravel.com/docs/6.x/artisan#tinker), or a really powerful CLI commands library. So this seem doable, and wanted to know whether people @directus/api already thought about this.

**Comment by next-direction** [comment]
This could be related to #31

**Comment by WoLfulus** [comment]
yeah, I've been thinking about this for more than a year now. at least for my workflows it would make it a lot easier for example to spawn new directus instances using a "template" project for new clients that needs something I already have done, or to make a "cloud" hosting of some kind of service. 

I already played with this for some time too (external cli for directus), I might have it here rotting somewhere.

Anyways, I'd love to talk about those if you're interested. I also like all of your suggestions for the codebase and I'm moving all of them to the laravel branch, if you're interested I'd love to get your feedback over slack, and we can keep talking. I'm there most of the time!

Just let me know!
